### ISSUE TITLE
Increasing the variable containing the size of files in FIM

### ISSUE THREAD

**Issue by jotacarma90** [wazuh/wazuh#24127]
# Description

Hi team, some time ago we were working on an issue to fix some problems in analysisd to process FIM events coming from large files (bigger than 2gb): https://github.com/wazuh/wazuh/issues/17184

This issue was solved by the analysisd (Linux) side without problems by modifying the variable ([PR](https://github.com/wazuh/wazuh/pull/17415/files)), which was an `int`, to an `unsigned long int`.

However, we have detected that although analysisd can correctly process these events, there are still problems in the syscheck part for Windows.
This is because in Linux, `unsigned long int` corresponds to 8 bytes (from 0 to 18446744073709551615 inclusive). However in the Windows compilation, this unsigned long int still contains only 4 bytes (0 to 4,294,967,295), so the problem appears in Windows for files larger than 4gb:
https://www.ibm.com/docs/en/ibm-mq/9.1?topic=platforms-standard-data-types-unix-linux-windows

## Tested on Windows 11

- Command:
`fsutil file createnew c:\test\filetest1 5000000000`
![image](https://github.com/wazuh/wazuh/assets/60003131/b8cfa62a-e0cc-4cca-8fc4-ed146b701137)

- This event correspond to file with 5.000.000.000 bytes (+4GB):
```
2024/06/17 00:39:35 wazuh-agent[6364] run_check.c:124 at send_syscheck_msg(): DEBUG: (6321): Sending FIM event: {"data":{"attributes":{"attributes":"ARCHIVE","checksum":"50627c9b0690670a359e3605ac02dd5e535345e0","hash_md5":"d41d8cd98f00b204e9800998ecf8427e","hash_sha1":"da39a3ee5e6b4b0d3255bfef95601890afd80709","hash_sha256":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","inode":0,"mtime":1718609975,"perm":{"S-1-5-32-544":{"name":"Administrators","allowed":["delete","read_control","write_dac","write_owner","synchronize","read_data","write_data","append_data","read_ea","write_ea","execute","read_attributes","write_attributes"]},"S-1-5-18":{"name":"SYSTEM","allowed":["delete","read_control","write_dac","write_owner","synchronize","read_data","write_data","append_data","read_ea","write_ea","execute","read_attributes","write_attributes"]},"S-1-5-32-545":{"name":"Users","allowed":["read_control","synchronize","read_data","read_ea","execute","read_attributes"]},"S-1-5-11":{"name":"Authenticated Users","allowed":["delete","read_control","synchronize","read_data","write_data","append_data","read_ea","write_ea","execute","read_attributes","write_attributes"]}},"size":705032704,"type":"file","uid":"S-1-5-32-544","user_name":"Administrators"},"mode":"realtime","path":"c:\\test\\filetest1","timestamp":1718609975,"type":"added","version":"2.0"},"type":"event"}
```
- `"size":705032704 ≈ 672MB`

## To do
- [x] We must make sure that FIM can correctly process this type of files. (unsigned long long int)

**Comment by jotacarma90** [comment]
# Update 01/07/2024

- Working on `bug/24127-windows-FIM-size-var` (currently local) modifiying every `unsigned long int size;` into `unsigned long long int size;`.
- Pending testing and check ITs related.

**Comment by jotacarma90** [comment]
# Update 02/07/2024

- After modifying the size variable from long int to long long long int, I still detected the incorrect operation.
- I have continued analyzing the code and I have seen that the function stat of Windows works with a struct stat that uses a long int for the size, with which the problem started from there.
- To solve it I had to modify both the function to use from stat to _stat64, and the struct also, from stat to _stat64.
- After that, I have found some problems of segfault by using in the snprintf %lu for the long long long, when it should be %llu.
- Currently working on the unit tests, which have been broken after these changes and are not being easy to repair.
- This issue is on hold because it is lower priority and has exceeded the estimated time for its priority. It will be finished another time.

**Comment by jotacarma90** [comment]
Since this fix is not so urgent for the current release, we are going to push it to the next release and remove it from 4.10.0.

**Comment by thefactor82** [comment]
Hi everyone,
@vikman90 asked me to answer on this issue, coming front this one: #20128 
I'm using wazuh 4.7.0 and the target machine where I see the problem with size of files is an Ubuntu 22.04 Linux virtual machine that we are using with samba shares used mainly by Windows and Mac PC.

**Comment by vikman90** [comment]
Hi @thefactor82,

I quickly changed the previous comment, realizing that the FIM implementation for Linux already has a fix for 4.9.0. Nevertheless, we will test this case.

Thanks for your comment.
Cheers!

**Comment by thefactor82** [comment]
I will plan an upgrade to my infr as soon as possible! Thanks
